App/View Controllers/Posts/PostsPageViewController.swift

```diff
@@ -1016,6 +1016,34 @@ final class PostsPageViewController: ViewController {
                     quotePost()
                 }
             }))
+        } else {
+            items.append(IconActionItem(.copyPost, block: {
+                func copyPost(_ post: Post) {
+                    let overlay = MRProgressOverlayView.showOverlayAdded(to: self.postsView.renderView,
+                                                                         title: LocalizedString("posts-page.copied-post"),
+                                                                         mode: .checkmark,
+                                                                         animated: true)
+                    overlay?.tintColor = self.theme["tintColor"]
+                 
+                    ForumsClient.shared.quoteBBcodeContents(of: post)
+                        .done { [weak self] bbcode in
+                            guard self != nil else { return }
+                            UIPasteboard.general.string = bbcode
+                        }
+                        .catch { [weak self] error in
+                            guard let self = self else { return }
+                            let alert = UIAlertController(title: LocalizedString("posts-page.error.could-not-copy-post"), error: error)
+                            self.present(alert, animated: true)
+                        }
+                        .finally {
+                            DispatchQueue.main.async {
```

Comment:
This isn't necessary, `finally` blocks run on the main queue.

Reply:
Haha I didn't know that.

This is why I raise PRs :)

Will fix

Reply:
This should now be correct in latest commit.